app/src/main/java/com/cronus/zdone/api/TasksRepositoryImpl.kt

```diff
@@ -1,78 +1,104 @@
 package com.cronus.zdone.api
 
 import com.cronus.zdone.AppExecutors
-import com.cronus.zdone.WorkTimeManager
 import com.cronus.zdone.api.model.*
 import com.cronus.zdone.home.HomeScreen
 import io.reactivex.Observable
+import io.reactivex.ObservableEmitter
+import java.util.*
 import javax.inject.Inject
 import javax.inject.Singleton
+import kotlin.collections.HashSet
 
 @Singleton
 class TasksRepositoryImpl @Inject constructor(
-        var appExecutors: AppExecutors,
-        var zdoneService: ZdoneService,
-        var workTimeManager: WorkTimeManager
+    var appExecutors: AppExecutors,
+    var zdoneService: ZdoneService
 ) : TasksRepository {
 
-    private val CACHE_REFRESH_TIME = 2 * 60 * 1_000_000_000L
+    private val CACHE_REFRESH_TIME = 2 * 60 * 1_000_000_000L // 2 mins in nanoseconds
 
-    var cachedData: Observable<Tasks>? = null
```

Comment:
the `Observable` nature of `cachedData` has been replaced by the observable `dataStream`